pylint/checkers/typecheck.py

```diff
@@ -1504,10 +1504,10 @@ def visit_call(self, node: nodes.Call) -> None:
         # includes an implicit `self` argument which is not present in `called.args`.
         if (
             isinstance(node.frame(), nodes.ClassDef)
-            and isinstance(node.parent, (nodes.Assign, nodes.AnnAssign))
```

Comment:
This `if` block was added by me originally. I don't think there was any need to specify that this node was an assignment; i.e. this should be checked also:
class X:
    def func(xx):
        ...
    func()